.github/workflows/prerelease.yml

```diff
@@ -59,6 +60,8 @@ jobs:
     steps:
       - name: "Check out repository code"
         uses: "actions/checkout@v2"
+      - name: "Check & Build Dependencies Container Image"
+        uses: "./.github/workflows/build_dependency_image_conditional.yml"
```

Comment:
Should `prerelease` and `release` perhaps be unconditional and always build the dependencies image?

Reply:
Thanks for the callout, jogged my thinking that while we tag off of `main`, what ends up being the "reference" in CI for releases is not the branch but tag, see: https://github.com/nautobot/nautobot/actions/runs/2994196181/jobs/4803333403#step:10:123

I'll have to poke around how to solve this.

Reply:
Tweaked in latest to publish `nautobot-dependencies:vX.Y.Z-py3.N` on every pre-/release.